test/ClientTest.cpp

```diff
@@ -1324,6 +1332,30 @@ test_wait(Connector<BUFFER, NetProvider> &client)
 	fail_unless(result.header.sync == static_cast<int>(f1));
 	fail_unless(result.header.code == 0);
 
+#ifdef __linux__
+	TEST_CASE("waitAny internal wait failure (gh-121)");
+	static constexpr double LONG_SLEEP_TIME = 2;
+	f = conn.call("remote_sleep", std::forward_as_tuple(LONG_SLEEP_TIME));
+	fail_unless(!conn.futureIsReady(f));
+	pthread_t tid = pthread_self();
+	struct sigaction act;
+	act.sa_handler = sigusr_handler;
+	act.sa_flags = 0;
+	sigemptyset(&act.sa_mask);
+	sigaction(SIGUSR1, &act, nullptr);
+	auto &&signal_thread = std::thread([tid] {
+		sleep(LONG_SLEEP_TIME / 2);
+		pthread_kill(tid, SIGUSR1);
+	});
+	fail_unless(!client.waitAny().has_value());
+	fail_unless(client.waitAny().has_value());
+	fail_unless(conn.futureIsReady(f));
+	response = conn.getResponse(f);
+	fail_unless(response.has_value());
```

Comment:
It would be nice to check that all wait functions handle errors if it's easy to do so.

Reply:
Done.